x-pack/plugins/apm/public/components/app/service_inventory/service_groups/index.tsx

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import {
+  EuiEmptyPrompt,
+  EuiFieldText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormControlLayout,
+  EuiText,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { isEmpty, sortBy } from 'lodash';
+import React, { useState, useCallback } from 'react';
+import { useHistory } from 'react-router-dom';
+import { useServiceGroupContext } from '../../../../context/service_group/use_service_group_context';
+import { useLocalStorage } from '../../../../hooks/use_local_storage';
+import { FETCH_STATUS, useFetcher } from '../../../../hooks/use_fetcher';
+import { ServiceGroupsList } from './service_groups_list';
+import { Sort } from './sort';
+import { useNavigateToServiceGroupInventory } from './use_navigate_to_service_group';
+import { fromQuery, toQuery } from '../../../shared/links/url_helpers';
+
+export type ServiceGroupsSortType = 'recently_added' | 'alphabetical';
+
+export function ServiceGroups() {
+  const { reloadServiceGroupList, serviceGroupName } = useServiceGroupContext();
+
+  const history = useHistory();
+  if (serviceGroupName) {
+    const { location } = history;
+    history.replace({
+      ...location,
+      search: fromQuery({
+        ...toQuery(location.search),
+        serviceGroup: undefined,
+      }),
+    });
+  }
+
+  const [filter, setFilter] = useState('');
+
+  const [apmServiceGroupsSortType, setServiceGroupsSortType] =
+    useLocalStorage<ServiceGroupsSortType>(
+      'apm.serviceGroupsSortType',
+      'recently_added'
+    );
+
+  const {
+    data = { serviceGroups: [] },
+    status,
+    refetch,
+  } = useFetcher(
+    (callApmApi) => callApmApi('GET /internal/apm/services/groups'),
+    []
+  );
+  reloadServiceGroupList.current = refetch;
```

Comment:
Can we do something pub/sub like

```suggestion
  reloadServiceGroupList.onClose(()=> { 
    refetch(); 
  })
```

?